File: tests/machine/x/fortran/partial_application.f90
program partial_application
  implicit none
  integer :: add5
  add5 = add(5)
  print *, add5(3)
  contains
  recursive integer function add(a,b) result(res)
    integer, intent(in) :: a
    integer, intent(in) :: b
    res = (a + b)
    return
  end function add
end program partial_application

File: tests/machine/x/cs/partial_application.cs
using System;

class Program {
    static int add(int a, int b) {
        return (a + b);
    }
    
    static void Main() {
        int add5 = new Func<int, int>((int p0) => { return add(5, p0); });
        Console.WriteLine(add5(3));
    }
}
